functions/vpn.bash

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+## install wireguard from unstable Debian
+## as long as it is not in the Raspbian repo
+##
+##   install_wireguard
+##
+install_wireguard() {
+  if [[ "$1" == "remove" ]]; then
+
+    apt remove --yes wireguard
+
+    systemctl stop wg-quick@wg0
+    rm -f /lib/systemd/system/wg-quick*
+    systemctl -q daemon-reload &>/dev/null
+
+    rm -f /etc/apt/sources.list.d/wireguard.list
+    apt_update
+    if [ -n "$INTERACTIVE" ]; then
+      whiptail --title "Wireguard VPN removed" --msgbox "We permanently removed the Wireguard installation from your box." 8 80 3>&1 1>&2 2>&3
+    fi
+    return 0
+  fi
+  if [[ "$1" != "install" ]]; then return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Installing Wireguard and enabling VPN remote access... "
+  echo "deb http://deb.debian.org/debian/ unstable main" > /etc/apt/sources.list.d/wireguard.list
+  apt-key adv --keyserver   keyserver.ubuntu.com --recv-keys 04EE7237B7D453EC
+  apt-key adv --keyserver   keyserver.ubuntu.com --recv-keys 648ACFD622F3D138
+
+  # important to avoid release mixing:
+  # prevent RPi from using the Debian distro for normal Raspbian packages
+  sh -c 'printf "Package: *\nPin: release a=unstable\nPin-Priority: 90\n" > /etc/apt/preferences.d/limit-unstable'
+  apt_update
+
+  # headers required for wireguard-dkms module to be built "live"
+  apt-get install --yes wireguard raspberrypi-kernel-headers
```

Comment:
Remember to implement `if [command]; then echo "OK"; else echo "FAILED"; fi` checks to ensure proper failure output on users end. See `packages.bash` for many great examples of what I mean